guardian/pressreader
# Add business logic to handler

## What does this change?

An initial pass at implementing the business logic for the pressreader job.

1. Take a config file that specifies a list of 'sections'.
2. Fetch the fronts collections and CAPI backfill searches for each section.
3. Fetch details for each article that was returned by step (2).
4. Check the articles for eligibility:
    - Exceeds a minimum wordcount.
    - Was published in the last 24 hours.
    - Doesn't include any 'banned' tags.
    - Hasn't already been included in another section of this edition.

It then returns a list of sections and article ids, to be saved to S3 by the handler function.

## How to test

- Set a CAPI api key in `resources.sh` (but make sure not to commit this!)
- Run `npm run dev`.
- Output should be generated for the example config in `config.ts`.

## How can we measure success?

This is part of migrating the existing service from on-prem to AWS Lambda.
